docs/adr/0002-shopping-lists-lineitem-and-textlineitem-update-actions.md

```diff
@@ -0,0 +1,336 @@
+                exactly same, we need to find a better way to avoid creating unnecessary actions.
+            </p>
+        </td>
+    </tr>
+    <tr>
+        <th colspan="2">Proposed solution</th>
+    </tr>
+    <tr>
+        <td colspan="2">
+            <p>
+                The solution idea about the new line item and changed order is still same with the case-1, but do we
+                need to remove and add line item with <b>SKU-1</b>? Which is not needed and we could start the removing
+                and adding from the changed order.
+            </p>
+            <pre lang="json">
+{
+  "version": 1,
+  "actions": [
+    {
+      "action": "removeLineItem",
+      "lineItemId": "24de3821-e27d-4ddb-bd0b-ecc99365285f"
+    },
+    {
+      "action": "addLineItem",
+      "sku": "SKU-3",
+      "quantity": 3,
+      "addedAt": "2020-11-05T10:00:10.101Z",
+      "custom": {
+        "type": {
+          "key": "custom-type-for-shoppinglists"
+        },
+        "fields": {
+          "textField": "text-3
+        }
+      }
+    }
+  ]
+}
+</pre>
+        </td>
+    </tr>
+</table>
+
+### How addedAt will be compared?
+
+In commercetools shopping lists API, there is no [update action](https://docs.commercetools.com/api/projects/shoppingLists#update-actions) 
+to change the `addedAt` field of the `LineItem` and `TextLineItem`, also in API it has a default value described as `Defaults to the current date and time.`, 
+when it's not set in the draft, so how to compare and update this field?
+
+**Proposed solution:**
+
+A sync option for excluding `addedAt` (based on the draft) comparisons, which defaults to `true`. The advantage of this, customers could decide themself sync `addedAt`. 
+The drawback is when the option set to `false` and a difference is found by our utils, to ensure the order we need to remove and add all line items.  
+
+> Note: `addedAt` data might be important if the customer has a front end that sorts the line items with the date.
+
+## Decision (not agreed yet with the team)
+
+<!-- The change that we're proposing or have agreed to implement. -->
+
+- Product variant will be matched by its SKU, if the SKU not set for a LineItemDraft, the draft will not be synced and an error callback will be triggered.
+> Note: In commercetools API, the product variant to be selected in the LineItemDraft can be specified either by its product ID plus variant ID or by its SKU.
+
+- When a [Change LineItems Order](https://docs.commercetools.com/api/projects/shoppingLists#change-lineitems-order">) action is needed 
+the line items will be removed and added back with the order provided in the `ShoppingListDraft`.
+
+## Consequences
+
+<!-- What becomes easier or more difficult to do and any risks introduced by the change that will need to be mitigated. -->
+
+- To ensure the order, we need to remove and add line items, which means a bigger payload, so performance overhead.
```

Comment:
So we will not use `Change LineItems Order` at all? Can you also write this consequence here?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/614/commits/baa0c2643702ff3dc99c17ebbf14322b8cd52a92